SaraAlert/SaraAlert
# SARAALERT-680: Display banner when multiple users have a record open

# Description
Jira Ticket: SARAALERT-680

Show a banner when multiple users have a monitoree record open.

NOTE: You must run the following in development mode to enable caching (which this PR requires): `rails dev:cache`

## (Feature) Demo/Screenshots

<img width="1488" alt="Screen Shot 2021-05-03 at 4 23 36 PM" src="https://user-images.githubusercontent.com/14923551/116929859-0312c680-ac2d-11eb-9f29-0ab460967891.png">

## Important Changes

`app/controllers/patients_controller.rb`
- Add load/unload caching

`app/javascript/components/patient/Viewers.js`
- Displays multiple viewer info on monitoree page

`app/javascript/tests/patient/Viewers.test.js`
- Tests for new Viewers component

`app/models/user.rb`
- Add can_see_viewers_of_a_patient role check

`app/views/patients/show.html.erb`
- Include new Viewers component on patient show

`config/routes.rb `
- Include on unload route

`package.json`
- Include unload react hook lib

`docker-compose.yml`
- Add dedicated redis cache container

# Checklists

**Submitter:**


@sgober  :


@tstrass  :


@jkufro  :
